queries.sql

```diff
@@ -8,3 +8,93 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg>10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE NOT name='Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+/*
+Inside a transaction update the animals table by setting the species column to unspecified. 
+Verify that change was made. Then roll back the change and verify that species columns went 
+back to the state before tranasction.
+*/
+BEGIN TRANSACTION;
+	UPDATE animals SET species='unspecified';
+	SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+/* -------------------------------------------------------------------------------------------------------------- */
+
+/*
+Inside a transaction:
+Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+Commit the transaction.
+Verify that change was made and persists after commit.
+*/
+BEGIN TRANSACTION;
+	UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+	UPDATE animals SET species='pokemon' WHERE name NOT LIKE '%mon';
```

Comment:
**Change Required**
- [ ] UPDATE the animals table by setting the species column to pokemon for all animals that don't have species already set. Currently this will set `pokemon` as long as it doesn't end with `mon`. But the requirement dictates that `all animals that don't have species already set`. Let us try and fix this.

Reply:
All right, I fixed that.